packages/shared/src/components/fields/CustomSwitch.tsx

```diff
@@ -68,60 +77,69 @@ export function CustomSwitch({
-        onChange={onToggle}
-      />
-      <ContentContainer
-        ref={leftRef}
-        className={checked && 'text-text-tertiary'}
-      >
-        {typeof leftContent === 'string'
-          ? leftContent
-          : React.cloneElement(leftContent as ReactElement, {
-              className: classNames(
-                (leftContent as ReactElement).props.className,
-                leftClasses,
-              ),
-            })}
-      </ContentContainer>
-      <ContentContainer
-        ref={rightRef}
-        className={!checked && 'text-text-tertiary'}
-      >
-        {typeof rightContent === 'string'
-          ? rightContent
-          : React.cloneElement(rightContent as ReactElement, {
-              className: classNames(
-                (rightContent as ReactElement).props.className,
-                rightClasses,
-              ),
-            })}
-      </ContentContainer>
-      <span className="absolute inset-0 my-auto h-7 rounded-10 bg-raw-cabbage-50 opacity-24 group-hover:opacity-32" />
-      <span
+    <SimpleTooltip content={tooltip?.content} placement={tooltip?.placement}>
+      <label
         className={classNames(
-          'absolute  top-0 z-1 h-full rounded-12',
-          styles.knob,
+          'group relative flex h-9 w-fit cursor-pointer font-bold typo-callout',
+          className,
         )}
-        style={{
-          width: `${width}px`,
-          transform: checked
-            ? `translateX(calc(100% + ${difference}px))`
-            : 'translateX(0)',
-        }}
-      />
-    </label>
+        htmlFor={inputId}
+      >
+        <input
+          id={inputId}
+          name={name}
+          type="checkbox"
+          className="absolute h-0 w-0 opacity-0"
+          checked={checked}
+          onChange={onToggle}
+          disabled={disabled}
+        />
+        <ContentContainer
+          ref={leftRef}
+          className={checked && 'text-text-tertiary'}
+        >
+          {typeof leftContent === 'string'
+            ? leftContent
+            : React.cloneElement(leftContent as ReactElement, {
+                className: classNames(
+                  (leftContent as ReactElement).props.className,
+                  leftClasses,
+                ),
+              })}
+        </ContentContainer>
+        <ContentContainer
+          ref={rightRef}
+          className={!checked && 'text-text-tertiary'}
+        >
+          {typeof rightContent === 'string'
+            ? rightContent
+            : React.cloneElement(rightContent as ReactElement, {
+                className: classNames(
+                  (rightContent as ReactElement).props.className,
+                  rightClasses,
+                ),
+              })}
+        </ContentContainer>
+        <span
+          className={classNames(
+            'absolute inset-0 my-auto h-7 rounded-10 opacity-24 group-hover:opacity-32',
+            disabled ? 'bg-surface-disabled' : 'bg-raw-cabbage-50',
```

Comment:
```suggestion
            disabled ? 'bg-surface-disabled' : 'bg-accent-cabbage-bolder',
```
We should avoid using raw colors at all cost :)

Reply:
yup, this was already there and just didn't think of changing, will try to keep this in mind for next things 🙌